web3/utils/signing.py

```diff
@@ -9,3 +43,101 @@ def signature_wrapper(message, version=b'E'):
         return preamble + size + message
     else:
         raise NotImplementedError("Only the 'Ethereum Signed Message' preamble is supported")
+
+
+def annotate_transaction_with_chain_id(txn_obj):
```

Comment:
This function name is confusing because it doesn't match what it seems like the function is doing.  I'd suggest renaming to something like:

* `get_transaction_obj_for_signing`
* `get_unsigned_transaction`

Reply:
I fought with the name for long enough that I just decided to change the abstraction boundary.

It now returns the hash of the signed transaction. It's called `hash_of_signed_transaction()`.